### PR TITLE
Add native stacktrace field for PlatformException

### PR DESCRIPTION
**Pull request by lidongze91**
## Description

Exposing an extra field for native stacktrace from PlatformException. This change would follow by further changes on Flutter Engine to give back native stacktrace information for both iOS&Android. This is essential for crash reporting, currently our client can only see the error message `Instance of 'PlatformException'` if the crash is coming from native platform.

## Related Issues

b/158148913

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/services/message_codec.dart [hunk 1]
```diff
@@ -118,8 +119,11 @@ class PlatformException implements Exception {
   /// Error details, possibly null.
   final dynamic details;
 
+  /// Native stacktrace for the error, possibly null.
```
Comment (cyanglaz) [inline packages/flutter/lib/src/services/message_codec.dart:122]: nits:
Can we explain more?
Like what's "Native stacktrace", where does user get dart stack trance, etc.

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 2]
```diff
@@ -567,8 +567,11 @@ class StandardMethodCodec implements MethodCodec {
     final dynamic errorCode = messageCodec.readValue(buffer);
     final dynamic errorMessage = messageCodec.readValue(buffer);
     final dynamic errorDetails = messageCodec.readValue(buffer);
+    /// TODO(libe): Mute the readValue check for stacktrace for now.
+    /// Remove this once flutter engine is ready with the stacktrace fields.
```
Comment (cyanglaz) [inline packages/flutter/lib/src/services/message_codecs.dart:571]: nits:
use // for TODOs
could you also link this TODO to an issue to track?

**Review by cyanglaz — COMMENTED** [review]
Can we add tests for this?

File: packages/flutter/lib/src/services/message_codecs.dart [hunk 3]
```diff
@@ -567,8 +567,11 @@ class StandardMethodCodec implements MethodCodec {
...
+    final dynamic errorStacktrace = (buffer.hasRemaining) ? messageCodec.readValue(buffer) as String : null;
```
Comment (mehmetf) [inline packages/flutter/lib/src/services/message_codecs.dart:572]: I don't think this has to be dynamic? Just `String?`.

**Comment by goderbauer** [comment]
The test failures look related to this change. @lidongze91  can you take a look at those?

**Comment by lidongze91** [comment]
> The test failures look related to this change. @lidongze91 can you take a look at those?

Resolved.

**Review by cyanglaz — APPROVED** [review]
LGTM

**Comment by hpoul** [comment]
probably a stupid question.. is there a reason it's a lowercase `stacktrace` from native platforms vs. dart's [stackTrace](https://api.dart.dev/stable/2.9.0/dart-core/Error/stackTrace.html) 🤔️

**Comment by christopherfujino** [comment]
@lidongze91 this was merged with failing presubmit checks. It's been reverted in https://github.com/flutter/flutter/pull/63431. Please ensure CI is green before merging, for more info https://github.com/flutter/flutter/wiki/Tree-hygiene.

**Pull request merged**